src/zenml/artifacts/utils.py

```diff
@@ -554,6 +564,103 @@ def load_artifact_from_response(artifact: "ArtifactVersionResponse") -> Any:
     )
 
 
+def verify_artifact_is_downloadable(
+    artifact: "ArtifactVersionResponse",
+    zen_store: Optional["BaseZenStore"] = None,
+) -> "BaseArtifactStore":
+    """Verify that the given artifact is downloadable.
+
+    Args:
+        artifact: The artifact to verify.
+        zen_store: The ZenStore to use for finding the artifact store.
+
+    Raises:
+        IllegalOperationError: If the artifact is too large to be archived.
+        KeyError: If the artifact store is not found or the artifact URI does
+            not exist.
+
+    Returns:
+        The artifact store.
+    """
+    if not artifact.artifact_store_id:
+        raise KeyError(
+            f"Artifact '{artifact.id}' cannot be downloaded because the "
+            "underlying artifact store was deleted."
+        )
+
+    artifact_store = _load_artifact_store(
+        artifact_store_id=artifact.artifact_store_id, zen_store=zen_store
+    )
+
+    if not artifact_store.exists(artifact.uri):
+        raise KeyError(f"The artifact URI '{artifact.uri}' does not exist.")
+
+    size = artifact_store.size(artifact.uri)
+    max_download_size = handle_int_env_var(
+        ENV_ZENML_SERVER_FILE_DOWNLOAD_SIZE_LIMIT,
+        DEFAULT_ZENML_SERVER_FILE_DOWNLOAD_SIZE_LIMIT,
+    )
```

Comment:
You should use a ServerConfiguration setting for this.

Reply:
Done